tests/utils.py

```diff
@@ -117,10 +117,36 @@ def tox_footer(envs=None, spaces=8):
 
 
 def prep_tox_output(output):
-    """Remove time info from tox output"""
-    result = re.sub(r" \((\d+\.\d+|\d+) seconds\)", "", output)
-    result = re.sub(r" ✔ in (\d+\.\d+|\d+) seconds", "", result)
-    return result
+    """Remove time info from tox 4 output.
+    Strip spaces around operators in tox 3 output."""
+    if TOX4:
+        output = re.sub(r" \((\d+\.\d+|\d+) seconds\)", "", output)
+        output = re.sub(r" ✔ in (\d+\.\d+|\d+) seconds", "", output)
+    else:
+        output = re.sub(" ((<|>)=?) ", r"\1", output)
+    return output
+
+
+def expand_tox(text):
+    """Replace [[TOX4:...]] and [[TOX3:...]] expressions,
+    empty liens are stripped if created."""
+    source_lines = text.splitlines()
+    out_lines = []
+    for line in source_lines:
```

Comment:
Only this loop is kinda cryptic. I'd add comments saying that the first if not line is there to preserve empty lines in the original output and the second if line is there to avoid adding more empty lines if one of the inline expressions creates one.